src/main/java/sorald/sonar/RuleVerifier.java

```diff
@@ -54,6 +83,78 @@ public static void verifyHasIssue(List<String> filesToScan, JavaFileScanner chec
      */
     @SuppressWarnings("UnstableApiUsage")
     public static void verifyNoIssue(String filename, JavaFileScanner check) {
-        MultipleFilesJavaCheckVerifier.verifyNoIssue(Arrays.asList(filename), check);
+        JavaCheckVerifier.newVerifier().onFile(filename).withCheck(check).verifyNoIssues();
+    }
+
+    private static InputFile toInputFile(File baseDir, String filename) {
+        // must append a separator to the basedir string as Sonar appends the filenames directly to
+        // it
+        final String baseDirStr = baseDir.toString() + File.separator;
+        try {
+            return new TestInputFileBuilder(baseDirStr, filename)
+                    .setContents(new String(Files.readAllBytes(Paths.get(filename)), UTF_8))
+                    .setCharset(UTF_8)
+                    .setLanguage("java")
+                    .build();
+        } catch (IOException e) {
+            throw new RuntimeException("failed to read file " + filename);
+        }
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private static JavaAstScanner createAstScanner(
+            SonarComponents sonarComponents, List<JavaFileScanner> checks) {
+        JavaAstScanner scanner = new JavaAstScanner(sonarComponents);
+        VisitorsBridge visitorsBridge =
+                new VisitorsBridge(
+                        checks,
+                        // TODO set the classpath to something reasonable
+                        Collections.emptyList(),
+                        sonarComponents,
+                        SymbolicExecutionMode.ENABLED);
+        // TODO we may want to set the version of the visitors bridge, as without setting it the
+        //      implementation defaults to the highest version supported by the parser (currently
+        //      Java 14)
+        //      visitorsBridge.setJavaVersion(new JavaVersionImpl(8));
+        scanner.setVisitorBridge(visitorsBridge);
+        return scanner;
+    }
+
+    private static SoraldSonarComponents createSonarComponents(File baseDir) {
+        // FIXME The SensorContextTester is an internal and unstable component in sonar,
+        //       we should implement our own SensorContext
+        SensorContextTester context = SensorContextTester.create(baseDir);
```

Comment:
Like the comment in the code says, we shouldn't be using the internal `SensorContextTester`, but it will serve for the time being.